lib/Twig/Extensions/Node/Trans.php

```diff
@@ -41,6 +41,10 @@ public function compile(Twig_Compiler $compiler)
 
         $function = null === $this->getNode('plural') ? 'gettext' : 'ngettext';
 
+        if (null !== $this->getNode('domain')) {
```

Comment:
Actually using `hasNode` would be the best choice. Then I can tell if domain exists at all.